runtime/shared_heap.c

```diff
@@ -201,16 +209,34 @@ static void pool_release(struct caml_heap_state* local,
   CAMLassert(pool->sz == sz);
   local->stats.pool_words -= POOL_WSIZE;
   local->stats.pool_frag_words -= POOL_HEADER_WSIZE + wastage_sizeclass[sz];
-  /* TODO: give free pools back to the OS. Issue #698 */
   caml_plat_lock(&pool_freelist.lock);
   pool->next = pool_freelist.free;
   pool_freelist.free = pool;
   caml_plat_unlock(&pool_freelist.lock);
 }
 
+static void free_pool_freelist(void)
+{
+  caml_plat_lock(&pool_freelist.lock);
+  /* free all but the first item on the freelist */
+  pool* o = pool_freelist.free;
+
+  if (o && o->next) {
```

Comment:
```suggestion
  if (o) {
```
You don't need this test.

Reply:
Resolved in https://github.com/ocaml/ocaml/pull/12193/commits/28ae570952d932c752aa13ca2174e765d70db385